packages/common/src/polyfills/sdk-shim/withModalProvider.tsx

```diff
@@ -11,14 +13,32 @@ import { ModalProvider } from '../console-dynamic-plugin-sdk/app/modal-support/M
  * the modal provider context.
  */
 export const withModalProvider = <T extends object>(Component: React.ComponentType<T>): React.FC<T> => {
-  function ModalProviderHoc(props) {
+  const [major, minor] = getReleaseVersion();
+  let ModalProviderHoc;
+
+  const ModalProviderHocPolyfill = (props) => {
     return (
       <ModalProvider>
         <Component {...props} />
       </ModalProvider>
     );
   }
 
+  const ModalProviderHocSDK = (props) => {
+    return (
+      <Component {...props} />
+    );
+  }
+
+  /**
+   * Test console version and fallback to poyfill on older versions
+   */
+  if (major === 0 || (major >= 4 && minor > 11)) {
```

Comment:
since this logic is repeated I would move it inside the hook. i.e.
```
const [requiresPolyfill, major, minor] = getReleaseVersion();
```

Reply:
nice !

Reply:
fixed